### DISCUSSION TITLE
Adding Array Elements Dynamically

### DISCUSSION THREAD

**Discussion by geethabathula** [Jay-study-nildana/FrontEndForStudents#27] [Coding Questions]
Hi @Jay-study-nildana ,

I was attempting to store data from input field into an array. and displaying the data entered onclick. Even showing the previous array data as well. I was not able to store prev data everytime on click the array is getting empty and only one value is being displayed.
Here is the link to pen [https://codepen.io/geethabathula/pen/qBJpmmY](https://codepen.io/geethabathula/pen/qBJpmmY)
Please help me solving this.

Thanks and Regards 
Geetha Supriya Bathula

**Comment by Jay-study-nildana** [comment]
@geethabathula 

First thing is, as there is no comments in the code, I am getting lost trying to understand what you have typed. So, as you grow and learn as a developer, please add some comments to your code, so others, like me, can understand what is happening.

Now, coming to the problem here. 

if you are thinking of storing things and then showing them, especially arrays, you should look at the following two topics. 

Topic 1

you should did into local storage. here are some links that can help you

1. https://codepen.io/jay-pancodu/pen/BaPOyvd
2. https://www.w3schools.com/jsref/prop_win_localstorage.asp

Topic 2

looks like you are getting serious about DOM manipulation. I would strongly recommend that you wait till we begin React JS (which we will be doing very soon in the batch). 

Another option is, if you want to do DOM (without React JS), then, I would like you to check out jQuery. 

1. https://jquery.com/

These days, we don't use jQuery that much, but, as you are exploring actively, it's not a bad idea to learn this library. 

Let me know if this solves you problem. If it does not, let me know and we will work together and solve this issue, one step at a time. 

Jay.
see my memes here, https://imgflip.com/user/codingtutorjay
join my open source community, here, https://github.com/BigbabeeSmallbabee
insta and twitter and other things, available on my github profile.

**Comment by geethabathula** [comment]
Hi Jay,
I have gone through topic 1 I was not able to understand anything.
Sure from next time will write comments.
Thanks and Regards 
Geetha Supriya Bathula

**Comment by geethabathula** [comment]
@Jay-study-nildana Please help in solving

**Reply by Jay-study-nildana** [reply]
@geethabathula 

My apologies for the delay, let me look into this. I dont know when I will fix this as you know, I am pathetic in CSS and HTML. I will do it. 

Jay.

**Comment by Vijay263601** [comment]
@geethabathula  I tried to encounter this problem but i dont know why it is showing this
![Screenshot 2023-05-17 104426](https://github.com/Jay-study-nildana/FrontEndForStudents/assets/125950839/257acc87-43fe-481e-a747-33a2f688acf8)

**Reply by Jay-study-nildana** [reply]
@Vijay263601 thanks for looking into this.

@geethabathula  see if you can take some help from @Vijay263601 I will also look into this issue, but I am hoping, it will be solved by Vijay :) 

Fingers crossed.

**Comment by Jay-study-nildana** [comment]
@geethabathula 

I have started doing some research into this. Let's solve this together. 

I have some questions.

1. What is this code supposed to do? Please give me an explanation about the goal of this code. 
2. Why are you using 'querySelector'. why not, getElementById ?
3. as you are dealing with array, why are you not using a list element in the HTML? I am not an expert in HTML and CSS, but basic logic tells me that when you are using an array, you should automatically have an empty list in HTML that will be used in your for loop. 
4. Also, you are accidentally triggering document.Location (and that is also linked to href), which is making it impossible to run the code.

So, once you give some answers, we can work together and rebuild this code from scratch. I will wait for your reply.

Thanks.

Jay.
